stress/src/main.rs

```diff
@@ -47,6 +49,214 @@ const DEFAULT_WRITE_REGION_COUNT: u64 = 5;
+        conflicts_with = "compact-ttl",
+        takes_value = true,
+        required = false,
+        about = "Compact log entries exceeding this threshold"
+    )]
+    compact_count: Option<u64>,
+
+    #[clap(
+        long = "force-compact-factor",
+        value_name = "factor",
+        takes_value = true,
+        default_value = "0.5",
+        validator = |s| {
+            let factor = s.parse::<f32>().unwrap();
+            if factor >= 1.0 {
+                Err(String::from("Factor must be smaller than 1.0"))
+            } else if factor <= 0.0 {
+                Err(String::from("Factor must be positive"))
+            } else {
+                Ok(())
+            }
+        },
+        about = "Factor to shrink raft log during force compact"
+    )]
+    force_compact_factor: f32,
+
+    #[clap(
+        long = "write-threads",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_WRITE_THREADS),
+        about = "Set the thread count for writing logs"
+    )]
+    write_threads: u64,
+
+    #[clap(
+        long = "write-ops-per-thread",
+        value_name = "ops",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_READ_OPS_PER_THREAD),
+        about = "Set the per-thread OPS for read entry requests"
+    )]
+    write_ops_per_thread: u64,
+
+    #[clap(
+        long = "read-thread",
+        value_name = "threads",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_READ_THREADS),
+        about = "Set the thread count for reading logs"
+    )]
+    read_threads: u64,
+
+    #[clap(
+        long = "read-ops-per-thread",
+        value_name = "ops",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_READ_OPS_PER_THREAD),
+        about = "Set the per-thread OPS for read entry requests"
+    )]
+    read_ops_per_thread: u64,
+
+    #[clap(
+        long = "entry-size",
+        value_name = "size",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_ENTRY_SIZE),
+        about = "Set the average size of log entry"
+    )]
+    entry_size: usize,
+
+    #[clap(
+        long = "write-entry-count",
+        value_name = "count",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_WRITE_ENTRY_COUNT),
+        about = "Set the average number of written entries of a region in a log batch"
+    )]
+    write_entry_count: u64,
+
+    #[clap(
+        long = "write-region-count",
+        value_name = "count",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_WRITE_REGION_COUNT),
+        about = "Set the average number of written regions in a log batch"
+    )]
+    write_region_count: u64,
+
+    #[clap(
+        long = "write-sync",
+        value_name = "sync",
+        takes_value = true,
+        default_value = formatcp!("{}", DEFAULT_WRITE_SYNC),
+        about = "Whether to sync write raft logs"
+    )]
+    write_sync: String,
```

Comment:
They can all be typed. Let's remove any `parse` from the code.

Reply:
changing `write_sync` and `reuse_data` to bool type will leading to compile error 'default_value is meaningless for bool'. it look like a clap problem same with structopt mentioned at https://github.com/TeXitoi/structopt/issues/93

Reply:
I see. That's because is a "is_present" flag. You can change it from `write_sync` to `write_without_sync` instead and remove default value setting.